### PR TITLE
Translations for CitizenEvents, Spigot and Paper

### PR DESCRIPTION
**Pull request by KyTDK**
Created spot for each .yml translation and modified code in java file to get language string

### CODE REVIEW

File: paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java [hunk 1]
```diff
@@ -209,10 +217,9 @@ private void onNPCClickEvent(NPCRightClickEvent event) { //Disconnect objectives
                     if (npcToEscort != null) {
                         if (npcToEscort.isSpawned() && (npcToEscort.getEntity().getLocation().distance(player.getLocation()) < 6)) {
                             activeObjective.addProgress(1, nqNPC);
-                            final String mmNpcName = main.getMiniMessage().serialize(LegacyComponentSerializer.legacyAmpersand().deserialize(npcToEscort.getName()));
 
                             player.sendMessage(main.parse(
-                                "<GREEN>You have successfully delivered the NPC <highlight>" + mmNpcName
+                                    main.getLanguageManager().getString("chat.npc.delivery-of-npc-complete", questPlayer)
```
Comment (AlessioGr) [inline paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java:222]: you forgot to add the map to replace the NPC name there

Comment (KyTDK) [reply paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java:222]: ah yes thank you will ammend

**Comment by AlessioGr** [comment]
Could you split this up into 3 PRs? Because the translations I could merge relatively quickly, the Integer[] additions I'd look at more closely and maybe add things to it myself there, and the config I'm not sure yet.

For the translations, can you move the objective-specific translations into their respective taskDescription config entry? E.g. delivery-progress I'd move there, just so all objective-specific strings are all together,
image
Or this would work too:
image

File: paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java [hunk 2]
```diff
@@ -228,7 +236,7 @@ private void onNPCClickEvent(NPCRightClickEvent event) { //Disconnect objectives
                             npcToEscort.despawn();
                         } else {
                             player.sendMessage(main.parse(
-                                "<RED>The NPC you have to escort is not close enough to you!"
+                                    main.getLanguageManager().getString("chat.npc.too-far", questPlayer)
```
Comment (AlessioGr) [inline paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java:239]: can you move this to the escortnpc objective?

File: paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java [hunk 3]
```diff
@@ -210,9 +218,9 @@ private void onNPCClickEvent(NPCRightClickEvent event) { //Disconnect objectives
                         if (npcToEscort.isSpawned() && (npcToEscort.getEntity().getLocation().distance(player.getLocation()) < 6)) {
                             activeObjective.addProgress(1, nqNPC);
                             final String mmNpcName = main.getMiniMessage().serialize(LegacyComponentSerializer.legacyAmpersand().deserialize(npcToEscort.getName()));
-
                             player.sendMessage(main.parse(
-                                "<GREEN>You have successfully delivered the NPC <highlight>" + mmNpcName
+                                    main.getLanguageManager().getString("chat.objectives.taskCompletion.deliverItems.delivery-of-npc-complete", questPlayer,
```
Comment (AlessioGr) [inline paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java:222]: can you move this to the escortnpc objective?

File: paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java [hunk 4]
```diff
@@ -196,7 +202,9 @@ private void onNPCClickEvent(NPCRightClickEvent event) { //Disconnect objectives
                     final String mmNpcName = main.getMiniMessage().serialize(LegacyComponentSerializer.legacyAmpersand().deserialize(npc.getName().replace("§","&")));
 
                     player.sendMessage(main.parse(
-                        "<GREEN>You talked to <highlight>" +mmNpcName
+                            main.getLanguageManager().getString("chat.talked-to-npc", questPlayer, Map.of(
```
Comment (AlessioGr) [inline paper/src/main/java/rocks/gravili/notquests/paper/events/hooks/CitizensEvents.java:205]: can you move this to the talktonpc objective?

File: plugin/src/main/resources/translations/en-US.yml [hunk 5]
```diff
@@ -86,6 +86,9 @@ chat:
     successfully-completed-rewards-prefix: '    <highlight>Rewards:'
     successfully-completed-rewards-rewardformat: '    <highlight> - <unimportant>%reward%'
     successfully-completed-rewards-suffix: <EMPTY>
+    taskCompletion:
+      deliverItems:
+        delivery-of-npc-complete: '<GREEN>You have successfully delivered the NPC <highlight> %NPCNAME%'
```
Comment (AlessioGr) [inline plugin/src/main/resources/translations/en-US.yml:91]: Move to taskDescription, just so objective stuff is all together in one specific place instead of spread out

File: paper/src/main/java/rocks/gravili/notquests/paper/structs/conditions/NumberCondition.java [hunk 6]
```diff
@@ -178,6 +178,12 @@ public String checkInternally(final QuestPlayer questPlayer) {
                 if (d <= numberRequirement) {
```
Comment (AlessioGr) [inline paper/src/main/java/rocks/gravili/notquests/paper/structs/conditions/NumberCondition.java:178]: Move to a separate PR, would like to do more work on that

**Review by AlessioGr — CHANGES_REQUESTED** [review]

**Comment by AlessioGr** [comment]
Closing due to inactivity
